### ISSUE TITLE
[typescript] Class-typed Types should use "implements Interface" if interface declarationKind is interface

### ISSUE THREAD

**Issue by alexeden** [dotansimha/graphql-code-generator#5643]
**Describe the bug**
If the `declarationKind` typescript plugin option is set such that `interface: interface` and `type: class`, the generated class of a type will use `extends InterfaceName`, which is not valid TS syntax.

**To Reproduce**

https://codesandbox.io/s/peaceful-swirles-1t0et?file=/schema.graphql

1. My GraphQL schema:

```graphql
interface Base {
    id: ID!
}

type MyType implements Base {
    id: ID!
}
```

2. My `codegen.yml` config file:

```yml
schema: schema.graphql
generates:
  types.ts:
    config:
      declarationKind:
        type: class
        interface: interface
    plugins:
      - typescript
```

**Expected behavior**

If classes are generated from types and interfaces are generated from interfaces, a type that implements an interface should look like:

```ts
export class MyType implements Base {
```

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/commit/e947f8e393e04dfd52f78bbb35b7e3e6af825c4f

**Comment by dotansimha** [comment]
Fixed in `@graphql-codegen/typescript@1.21.1`
